sdk/client/go/flipt-client-go/models.go

```diff
@@ -25,3 +25,15 @@ type BooleanEvaluationResponse struct {
 	RequestDurationMillis float64 `json:"request_duration_millis"`
 	Timestamp             string  `json:"timestamp"`
 }
+
+type VariantResult struct {
+	Status       string                     `json:"status"`
+	Result       *VariantEvaluationResponse `json:"result,omitempty"`
+	ErrorMessage string                     `json:"error_message,omitempty"`
+}
+
+type BooleanResult struct {
+	Status       string                     `json:"status"`
+	Result       *BooleanEvaluationResponse `json:"result,omitempty"`
+	ErrorMessage string                     `json:"error_message,omitempty"`
+}
```

Comment:
Take it or leave it:
```suggestion
type Result[R any] struct {
	Status       string `json:"status"`
	Result       *R     `json:"result,omitempty"`
	ErrorMessage string `json:"error_message,omitempty"`
}

type VariantResult Result[VariantEvaluationResponse]

type BooleanResult Result[BooleanEvaluationResponse]
```

Reply:
look at you crab dev 🦀

Reply:
😂 This is Go Mark

Reply:
i forgot Go had generics..

Reply:
😭 nice will do.

Reply:
I'll rebase and get this change in on Node PR